### PR TITLE
Toolset to migrate histrory data between OrientDB and MySQL

### PR DESCRIPTION
**Pull request by surabujin**
Python package with history migration toolset.

### CODE REVIEW

**Review by dpoltavets — COMMENTED** [review]
Should we add files from `tools/history-migration/src/kilda/__pycache__/` and `tools/history-migration/src/kilda/history_migration/__pycache__/` to `.gitignore`?

**Comment by surabujin** [comment]
PR have been splitted on 2 chunks - first one stays here(migration tool), second was moved to the new PR #4380

File: docker/db-migration/migrations/010-enforce-definite-types-on-history-objects.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  - changeSet:
+      id: tag
+      author: surabujin
+      changes:
+        - tagDatabase:
+            tag: 010-enforce-definite-types-on-history-objects
+
+  - changeSet:
+      id: add_unique_index_for_port_history
+      author: suerabujin
```
Comment (rozdy) [inline docker/db-migration/migrations/010-enforce-definite-types-on-history-objects.yaml:12]: typo

File: docker/mysql/setup_data/initial-schema.sql [hunk 2]
```diff
@@ -0,0 +1,58 @@
+-- initial mysql schema
+
+CREATE TABLE flow_events_history
```
[the remark this answers is not in the document]
Comment (rozdy) [reply docker/mysql/setup_data/initial-schema.sql:3]: I am ok with `history` suffix here but I think better use singular form for `event`

File: docker/mysql/setup_data/initial-schema.sql [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    unstructured json     not null,
+    time_create  datetime not null default now(),
+    time_modify  datetime,
+    event_time   datetime,
+    index        `ix_event_time` (event_time),
+    index        `ix_flow_id` (flow_id(64)),
+    unique index `ix_task_id` (task_id(64))
+);
+
+CREATE TABLE flow_event_actions
```
[the remark this answers is not in the document]
Comment (rozdy) [reply docker/mysql/setup_data/initial-schema.sql:19]: `actions` -> `action`

File: docker/mysql/setup_data/initial-schema.sql [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    time_create   datetime not null default now(),
+    time_modify   datetime,
+    event_time    datetime,
+    index         `ix_flow_event_id` (flow_event_id),
+    foreign key `fk_flow_event_actions_2_flow_event_history` (flow_event_id)
+        REFERENCES flow_events_history(id)
+        ON UPDATE CASCADE ON DELETE RESTRICT
+);
+
+CREATE TABLE flow_event_dumps
```
[the remark this answers is not in the document]
Comment (rozdy) [reply docker/mysql/setup_data/initial-schema.sql:34]: `dumps` -> `dump`

File: docker/mysql/setup_data/initial-schema.sql [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+(
+    id            integer primary key auto_increment,
+    flow_event_id integer     not null,
+    kind          varchar(64) not null,
+    unstructured  json        not null,
+    time_create   datetime    not null default now(),
+    time_modify   datetime,
+    index         `ix_flow_event_id` (flow_event_id),
+    foreign key `fk_flow_event_actions_2_flow_event_history` (flow_event_id)
```
[the remark this answers is not in the document]
Comment (rozdy) [reply docker/mysql/setup_data/initial-schema.sql:43]: Still wrong foreign key name

File: docker/mysql/setup_data/initial-schema.sql [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        REFERENCES flow_events_history (id)
+        ON UPDATE CASCADE ON DELETE RESTRICT
+);
+
+CREATE TABLE port_events_history
```
[the remark this answers is not in the document]
Comment (rozdy) [reply docker/mysql/setup_data/initial-schema.sql:48]: `events` -> `event`

File: tools/history-migration/setup.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+setup_deps = ['wheel']
+
+with open(path(
+        'src', 'kilda', 'history_migration', 'APP_META.json'), 'rt') as f:
+    meta = json.load(f)
+
+setup(
+    name=meta['name'],
+    version=meta['version'],
+    description='Monitors OpenKilda switch connections list to the speakers',
```
Comment (rozdy) [inline tools/history-migration/setup.py:59]: Wrong description

**Review by rozdy — APPROVED** [review]

**Review by niksv — APPROVED** [review]
is it ok that `tools/history-migration/requirements-dev.txt` is empty?

File: tools/history-migration/src/kilda/history_migration/cli.py [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+import pyorient
+
+from . import orphaned
+from . import pull_mysql
+from . import pull_orient
+from . import push
+
+
+@click.group()
+def cli():
```
Comment (timofei-durakov) [inline tools/history-migration/src/kilda/history_migration/cli.py:29]: what is this used for?

File: tools/history-migration/src/kilda/history_migration/mapper.py [hunk 9]
```diff
@@ -0,0 +1,534 @@
...
+    time_create = datetime_of_string(time_create)
+    time_modify = datetime_of_string(time_modify)
+    return model.FlowEventAction(
+        action_time, action_name, details,
+        time_create=time_create, time_modify=time_modify)
+
+
+# -- flow event dump --
+
+def _flow_event_dump_to_dump(dump):
```
Comment (timofei-durakov) [inline tools/history-migration/src/kilda/history_migration/mapper.py:149]: I'd add validation like 
```
if not dump:
    return {}  
```

**Review by timofei-durakov — APPROVED** [review]

File: tools/history-migration/src/kilda/history_migration/cli.py (hunk 8 above)
Comment (surabujin) [reply tools/history-migration/src/kilda/history_migration/cli.py:29]: Is path of "click" api, required to create command/options group.

File: confd/templates/base-storm-topology/topology.properties.tmpl [hunk 10]
```diff
@@ -55,6 +57,11 @@ opentsdb.metric.prefix = {{ getv "/kilda_opentsdb_metric_prefix" }}
 
 statistics.interval = {{ getv "/kilda_statistics_interval" }}
 
+hibernate.user = {{ getv "/kilda_hibernate_user" }}
+hibernate.password = {{ getv "/kilda_hibernate_password" }}
+hibernate.url = {{ getv "/kilda_hibernate_url" }}
+#hibernate.driver_class = com.mysql.jdbc.Driver
```
[the remark this answers is not in the document]
Comment (surabujin) [reply confd/templates/base-storm-topology/topology.properties.tmpl:63]: It shows "default" value of this option.

**Pull request merged**